distribution/integration-tests/src/test/java/org/jboss/tools/rsp/itests/wildfly/WildFlyPublishingTest.java

```diff
@@ -20,117 +21,268 @@
+		Attributes attrs = serverProxy.listDeploymentOptions(handle).get();
+		assertNotNull(attrs);
+	}
+
+	/*
+	 * Breaks the server, see
+	 * https://github.com/redhat-developer/rsp-server/issues/376
+	 */
+	@Ignore
+	@Test
+	public void testListDeploymentOptionsInvalidAttributes() throws InterruptedException, ExecutionException {
+		Attributes attrs = serverProxy.listDeploymentOptions(new ServerHandle("foo.server.id",
+				new ServerType("some.id", "my.server", "Random server type definition"))).get();
+		assertNull(attrs);
+	}
+
+	/*
+	 * Breaks the server, see
+	 * https://github.com/redhat-developer/rsp-server/issues/376
+	 */
+	@Ignore
+	@Test
+	public void testListDeploymentOptionsNullAttributes() throws InterruptedException, ExecutionException {
+		Attributes attrs = serverProxy.listDeploymentOptions(null).get();
+		assertNull(attrs);
+	}
+
+	/*
+	 * Breaks the server, see
+	 * https://github.com/redhat-developer/rsp-server/issues/376
+	 */
+	@Ignore
+	@Test
+	public void testGetDeployablesInvalidAttributes() throws InterruptedException, ExecutionException {
+		List<DeployableState> states = serverProxy.getDeployables(new ServerHandle("foo.server.id",
+				new ServerType("some.id", "my.server", "Random server type definition"))).get();
+		assertTrue(states.isEmpty());
+	}
+
+	/*
+	 * Breaks the server, see
+	 * https://github.com/redhat-developer/rsp-server/issues/376
+	 */
+	@Ignore
+	@Test
+	public void testGetDeployablesNullAttributes() throws InterruptedException, ExecutionException {
+		List<DeployableState> states = serverProxy.getDeployables(null).get();
+		assertFalse(states.isEmpty());
+	}
+
+	@Test
+	public void testGetDeployables() throws Exception {
+		DeployableReference ref2 = new DeployableReference("other_id", WAR_FILENAME);
+		serverProxy.addDeployable(new ServerDeployableReference(handle, ref2));
+		waitForDeployablePublishState(ServerManagementAPIConstants.PUBLISH_STATE_ADD, 10, client);
+		List<DeployableState> states = serverProxy.getDeployables(handle).get();
+		assertEquals(2, states.size());
+		DeployableReference actualRef = states.get(0).getReference();
+		DeployableReference actualRef2 = states.get(1).getReference();
+		assertTrue(actualRef.equals(reference) && !actualRef.equals(ref2)
+				|| actualRef.equals(ref2) && !actualRef.equals(reference));
+		assertTrue(actualRef2.equals(ref2) && !actualRef2.equals(reference)
+				|| actualRef2.equals(reference) && !actualRef2.equals(ref2));
+	}
+
+	@Test
+	public void testGetDeployablesEmptyDeployment() throws Exception {
+		waitForDeployablePublishState(ServerManagementAPIConstants.PUBLISH_STATE_ADD, 10, client);
+		serverProxy.removeDeployable(new ServerDeployableReference(handle, reference));
+		List<DeployableState> states = serverProxy.getDeployables(handle).get();
+		assertTrue(states.isEmpty());
+	}
+
+	protected void sendPublishRequest(int publishType) throws InterruptedException, ExecutionException {
```

Comment:
Would any of these protected methods come in handy in the deployment test? If so, move them up a level. Otherwise just make them private.

Reply:
you are right, moving up and reusing in quickstarts deploying test.